scripts/test-infra/test-e2e-presubmit.sh

```diff
@@ -1,67 +1,25 @@
 #!/bin/bash -e
 set -o pipefail
 
-# Configure environment
-KIND_IMAGE="kindest/node:v1.25.3"
-export IMAGE_REGISTRY="localhost:5001"
 export CLUSTER_NAME=$(git describe --tags --dirty --always)
 export KUBECONFIG="/tmp/kubeconfig_$CLUSTER_NAME"
 
-# create registry container unless it already exists
-reg_name='kind-registry'
-reg_port='5001'
-if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)" != 'true' ]; then
-  docker run \
-    -d --restart=always -p "127.0.0.1:${reg_port}:5000" --name "${reg_name}" \
-    registry:2
-fi
+minikube start --bootstrapper=kubeadm \
+  --vm-driver=docker \
+  --memory 8192 \
+  --cpus 4 \
```

Comment:
maybe we want to check `nproc`

Reply:
I will check it. Never used this utility before.
But actually, these resources is something I wanted to ask. IINW, minikube defaults to 2 CPU and 2GB memory, but I used 4 CPUs here and 8 GB memory for each cluster, which might cause resource saturation on the host when there are many of those clusters in parallel. So, I should admit that I have not verified if we really need 8 GB, I can perhaps do tests with 4 GB as well.

Reply:
agree, for an e2e pre test, 4 cpu's is a bit too much

Reply:
Check this and I think we could merge this one

Reply:
Agree, let's go with default resources, the rest of the PR looks ok

Reply:
based on the tests results 2GB, 2CPU is enough. But there are some other bits I would like to add on this PR. As such, will hold it and ping you folks once ready again.
/hold